internal/pkg/consensus/election.go

```diff
@@ -62,12 +62,22 @@ func (em ElectionMachine) GenerateWinningPoSt(ctx context.Context, allSectorInfo
 	}
 	poStRandomness := abi.PoStRandomness(randomness)
 
-	challengeIndexes, err := ffiwrapper.ProofVerifier.GenerateWinningPoStSectorChallenge(ctx, allSectorInfos[0].RegisteredProof, minerID, poStRandomness, uint64(len(allSectorInfos)))
+	rp, err := allSectorInfos[0].RegisteredProof.RegisteredWinningPoStProof()
+	if err != nil {
+		return nil, err
+	}
+	challengeIndexes, err := ffiwrapper.ProofVerifier.GenerateWinningPoStSectorChallenge(ctx, rp, minerID, poStRandomness, uint64(len(allSectorInfos)))
 	if err != nil {
 		return nil, err
 	}
 	challengedSectorInfos := filterSectorInfosByIndex(allSectorInfos, challengeIndexes)
```

Comment:
A pretty important scaling follow-up is to remove the allSectorInfos parameter and load sector info only for the very few sectors actually challenged. Please file an issue.

Reply:
Thank you so much for flagging this: #4018